app/uk/gov/hmrc/tradergoodsprofiles/connectors/BaseConnector.scala

```diff
@@ -30,6 +31,16 @@ trait BaseConnector {
       s"$routerBaseRoute/$eoriNumber/records/$recordId"
     )
 
+  def routerRouteGetRecords(
+    eoriNumber: String,
+    lastUpdatedDate: Option[String] = None,
+    page: Option[Int] = None,
+    size: Option[Int] = None
+  ): String = {
+    val uri = uri"$routerBaseRoute/$eoriNumber?lastUpdatedDate=$lastUpdatedDate&page=$page&size=$size"
```

Comment:
You may want to formattthe lastUpdayeDate to take time time in seconds (2024-05-15T15:23:12Z) as at this moment takes milliseconds. This is shown by the tests.